File: CS/CustomCopySelectionCommand.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.Commands;
using DevExpress.Office.Utils;
using DevExpress.XtraRichEdit.API.Native;
using System.Text;

namespace RichEditCustomCopyPaste {
    public class CustomCopySelectionCommand : CopySelectionCommand {
        public CustomCopySelectionCommand(IRichEditControl control)
            : base(control) {
        }

        private void Iterate(IDocumentVisitor visitor, DocumentRange documentRange)
        {
            DocumentIterator iterator = new DocumentIterator(documentRange, true);
            while (iterator.MoveNext())
                iterator.Current.Accept(visitor);
        }
       
        protected override void ExecuteCore() {
            RichEditControl richEditControl = (RichEditControl)Control;
            DocumentRange range = richEditControl.Document.Selection;

            PlainTextVisitor visitor = new PlainTextVisitor();
            Iterate(visitor, range);
            string plainText = visitor.Text;

            DataObject data = new DataObject();
            data.SetData(OfficeDataFormats.UnicodeText, plainText);
            data.SetData(OfficeDataFormats.Rtf, richEditControl.Document.GetRtfText(range));
            Clipboard.Clear();
            Clipboard.SetDataObject(data, false);
        }
    }

    public class PlainTextVisitor : DocumentVisitorBase
    {
        readonly StringBuilder buffer;
        public string Text { get { return buffer.ToString(); } }

        public PlainTextVisitor()
        {
            this.buffer = new StringBuilder();
        }
        public override void Visit(DocumentTableCellBorder cellBorder)
        {
            bool isFirstInRow = cellBorder.TableCellBorderProperties.IsFirstInRow;
            bool isLastInTable = cellBorder.TableCellBorderProperties.IsLastInTable;
            if (!isFirstInRow && !isLastInTable)
            {
                buffer.Length -= 2;
                buffer.Append("\t");
            }
            
            base.Visit(cellBorder);
        }
        public override void Visit(DocumentParagraphEnd paragraphEnd)
        {
            buffer.AppendLine();
            base.Visit(paragraphEnd);
        }
        public override void Visit(DocumentText text)
        {
            buffer.Append(text.Text);
            base.Visit(text);
        }
    }

}

File: VB/CustomCopySelectionCommand.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.Commands
Imports DevExpress.Office.Utils
Imports DevExpress.XtraRichEdit.API.Native
Imports System.Text

Namespace RichEditCustomCopyPaste
    Public Class CustomCopySelectionCommand
        Inherits CopySelectionCommand

        Public Sub New(ByVal control As IRichEditControl)
            MyBase.New(control)
        End Sub

        Private Sub Iterate(ByVal visitor As IDocumentVisitor, ByVal documentRange As DocumentRange)
            Dim iterator As New DocumentIterator(documentRange, True)
            Do While iterator.MoveNext()
                iterator.Current.Accept(visitor)
            Loop
        End Sub

        Protected Overrides Sub ExecuteCore()
            Dim richEditControl As RichEditControl = CType(Control, RichEditControl)
            Dim range As DocumentRange = richEditControl.Document.Selection

            Dim visitor As New PlainTextVisitor()
            Iterate(visitor, range)
            Dim plainText As String = visitor.Text

            Dim data As New DataObject()
            data.SetData(OfficeDataFormats.UnicodeText, plainText)
            data.SetData(OfficeDataFormats.Rtf, richEditControl.Document.GetRtfText(range))
            Clipboard.Clear()
            Clipboard.SetDataObject(data, False)
        End Sub
    End Class

    Public Class PlainTextVisitor
        Inherits DocumentVisitorBase

        Private ReadOnly buffer As StringBuilder
        Public ReadOnly Property Text() As String
            Get
                Return buffer.ToString()
            End Get
        End Property

        Public Sub New()
            Me.buffer = New StringBuilder()
        End Sub
        Public Overrides Sub Visit(ByVal cellBorder As DocumentTableCellBorder)
            Dim isFirstInRow As Boolean = cellBorder.TableCellBorderProperties.IsFirstInRow
            Dim isLastInTable As Boolean = cellBorder.TableCellBorderProperties.IsLastInTable
            If Not isFirstInRow AndAlso Not isLastInTable Then
                buffer.Length -= 2
                buffer.Append(vbTab)
            End If

            MyBase.Visit(cellBorder)
        End Sub
        Public Overrides Sub Visit(ByVal paragraphEnd As DocumentParagraphEnd)
            buffer.AppendLine()
            MyBase.Visit(paragraphEnd)
        End Sub
        Public Overrides Sub Visit(ByVal text As DocumentText)
            buffer.Append(text.Text)
            MyBase.Visit(text)
        End Sub
    End Class

End Namespace